jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/lambda/nodejs/NodeJsSamProjectWizard.kt

```diff
@@ -0,0 +1,78 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.nodejs
+
+import com.intellij.javascript.nodejs.interpreter.NodeJsInterpreter
+import com.intellij.javascript.nodejs.interpreter.NodeJsInterpreterField
+import com.intellij.javascript.nodejs.interpreter.NodeJsInterpreterManager
+import com.intellij.javascript.nodejs.interpreter.NodeJsInterpreterRef
+import com.intellij.lang.javascript.dialects.JSLanguageLevel
+import com.intellij.lang.javascript.settings.JSRootConfiguration
+import com.intellij.openapi.project.ProjectManager
+import com.intellij.openapi.roots.ModifiableRootModel
+import com.intellij.openapi.ui.ValidationInfo
+import software.amazon.awssdk.services.lambda.model.Runtime
+import software.aws.toolkits.jetbrains.services.lambda.SamNewProjectSettings
+import software.aws.toolkits.jetbrains.services.lambda.SamProjectTemplate
+import software.aws.toolkits.jetbrains.services.lambda.SamProjectWizard
+import software.aws.toolkits.jetbrains.services.lambda.SdkSettings
+import software.aws.toolkits.jetbrains.ui.wizard.SamProjectGenerator
+import software.aws.toolkits.jetbrains.ui.wizard.SdkSelectionPanel
+import software.aws.toolkits.resources.message
+import javax.swing.JComponent
+import javax.swing.JLabel
+
+class NodeJsSamProjectWizard : SamProjectWizard {
+    override fun createSdkSelectionPanel(generator: SamProjectGenerator): SdkSelectionPanel =
+        NodeJsSdkSelectionPanel()
+
+    override fun listTemplates(): Collection<SamProjectTemplate> = listOf(
+        SamHelloWorldNodeJs()
+    )
+}
+
+class NodeJsSdkSelectionPanel : SdkSelectionPanel {
+    private var interpreterPanel: NodeJsInterpreterField? = null
+
+    override val sdkSelectionLabel: JLabel?
+        get() = JLabel("Node interpreter:")
+
+    override val sdkSelectionPanel: JComponent
+        get() {
+            val project = ProjectManager.getInstance().defaultProject
+            val nodeJsInterpreterField = object : NodeJsInterpreterField(project, false) {
+                override fun isDefaultProjectInterpreterField(): Boolean = true
+            }
+            nodeJsInterpreterField.interpreterRef = NodeJsInterpreterManager.getInstance(project).interpreterRef
+            interpreterPanel = nodeJsInterpreterField
+            return nodeJsInterpreterField
+        }
+
+    override fun registerListeners() { }
+
+    override fun validateAll(): List<ValidationInfo>? = null
+
+    override fun getSdkSettings(): SdkSettings = NodeJsSdkSettings(interpreter = interpreterPanel?.interpreter)
+}
+
+class NodeJsSdkSettings(
+    var interpreter: NodeJsInterpreter? = null,
+    var languageLevel: JSLanguageLevel = JSLanguageLevel.ES6
```

Comment:
Nothing can modify this, so why have it?

Reply:
IntelliJ is using `ES5` by default. We want to override it since our template is using es6 grammars. For a long term, I think the template should tell us what minimum language level it supports so that we can set this value based on the that.

Reply:
Well even then why is it not a val?

Reply:
Changed to `val`